### PR TITLE
[dont merge] support both apigw and alb events as http::{Request,Response} (cargo features)

### PR DESCRIPTION
**Pull request by softprops**
*Issue #, if available:*

*Description of changes:*

bizzaro branch off  of https://github.com/awslabs/aws-lambda-rust-runtime/pull/64 with cargo feature flag toggles for conditional compilation

Have not checked the documentation checkbox yet because Im not sure this will look like the end state

By submitting this pull request

- [X] I confirm that my contribution is made under the terms of the Apache 2.0 license.
- [ ] I confirm that I've made a best effort attempt to update all relevant documentation.

### CODE REVIEW

File: lambda-http/src/request.rs [hunk 1]
```diff
@@ -21,50 +23,135 @@ use crate::{
...
+        api_id: String,
+        identity: Identity,
+    },
+    /// ALB request context
+    #[serde(rename_all = "camelCase")]
+    Alb { elb: Elb },
+}
+
+/// ALB request context
+#[cfg(all(feature = "alb", not(all(feature = "apigw", feature = "alb"))))]
```
Comment (softprops) [inline lambda-http/src/request.rs]: TIL learned cfg attrs are like a smaller dialect of lisp!

File: lambda-http/Cargo.toml [hunk 2]
```diff
@@ -30,3 +30,8 @@ serde_urlencoded = "0.5"
 [dev-dependencies]
 log = "^0.4"
 simple_logger = "^1"
+
+[features]
+default = ['apigw', 'alb']
```
Comment (softprops) [inline lambda-http/Cargo.toml]: big drawback here. if a consumer sets default-flags to false you'll get neither apigw nor alb! I don't think there's a way to express "there can be at least one!" in cargo

**Comment by softprops** [comment]
gonna fix up these merge conflicts...

**Comment by softprops** [comment]
with the 0.2 release looming what do you think about fold on this for now until we find its actually useful to enable feature flagging these. it seems the cost of maintenance added may not be worth the benefit. I'm kind of curious about hearing if this is an actual pain point _before_ investing in an ointment to relieve it. Thoughts?

**Comment by davidbarsky** [comment]
> I'm kind of curious about hearing if this is an actual pain point before investing in an ointment to relieve it. Thoughts?

I think that's a good point. Feature flagging it _seems_ cleaner, but it might not be worth the complexity.

**Comment by softprops** [comment]
Cool. I'll let you folks make the call. I'd be fine to close this with commitment to revisit this if there's demand for it

**Comment by softprops** [comment]
I havent seen any demand for this so I'm gonna close this for now to make sure the pull queue is only populated with pulls that spark joy

**Pull request closed without merging**